provider/azure/config.go

```diff
@@ -149,6 +149,17 @@ func getCloudConfiguration(name string) (cloud.Configuration, error) {
 		return cloud.AzureGovernment, nil
 	case "AZURECHINACLOUD":
 		return cloud.AzureChina, nil
+	case "AZURECUSTOMCLOUD":
+		azureAdEndpoint := os.Getenv("AZURE_AD_ENDPOINT")
```

Comment:
I'm not sure this is the best way to get this. This method (getCloudConfiguration) gets called with the cloud type (e.g. "AZURECLOUD"), which comes from the config file but then we take the endpoint config from some environment variable. I think it could be better to have the endpoint config as part of the config file

Reply:
@Jeremy-Boyle we are waiting for this one to comment or fix